core/extraction/function_extractor.py

```
@@ -217,6 +166,65 @@
+                                f"Extracted function: {extracted_function.name}, Arguments: {[arg.name for arg in extracted_function.args]}, Return Type: {extracted_function.returns['type']}"
+                            )
+                            if self.context.metrics_collector:
+                                self.context.metrics_collector.update_scan_progress(
+                                    self.context.module_name
+                                    or Path(
+                                        getattr(self.context.base_path, "name", "")
+                                    ).stem,
+                                    "function",
+                                    node.name,
+                                )
+                    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)